api/src/main/java/edu/wpi/first/shuffleboard/api/sources/recording/Recording.java

```diff
@@ -30,53 +41,110 @@ public void append(TimestampedData data) {
       if (last == null || data.getTimestamp() > last.getTimestamp()) {
         last = data;
       }
+    } finally {
+      lock.writeLock().unlock();
     }
   }
 
-  @SuppressWarnings("JavadocMethod")
+  /**
+   * Adds a marker to this recording.
+   *
+   * @param marker the marker to add
+   */
+  public void addMarker(Marker marker) {
+    ThreadUtils.withLock(lock.writeLock(), () -> markers.add(marker));
+  }
+
+  /**
+   * Gets the markers in this recording. Note: if you need to get the data objects and markers at the same time,
+   * use {@link #copyData copyData()} to avoid locking issues (the individual methods each lock, but the lists may
+   * change between method calls).
+   *
+   * <p>In live recordings, this list will not be exhaustive. Markers are removed after saving to disk to reduce
+   * memory and CPU usage. Recordings loaded in playback mode will have all the markers present.
+   *
+   * @return the markers in this recording
+   */
+  public List<Marker> getMarkers() {
+    return ThreadUtils.withLock(lock.readLock(), () -> markersReadOnly);
```

Comment:
Possible concurrent modification exception here when iterating and a different thread calls `clear`.